### PR TITLE
feat(ft-update-trip-status): update trip status

### PR DESCRIPTION
**Pull request by Patience250**
## # What does this PR do?
 Update trip status 
### Description of tasks to be completed
- change status from pending to rejected or approved
- protect routes for changing trip status
### How to test manually
- Clone the repository 
- Run npm install
- Run npm run dev command
- In postman send a put request  to ``localhost:4000/api/trips/status/:tripId'`` to update``trip status``

### CODE REVIEW

File: src/middlewares/trip.js [hunk 1]
```diff
@@ -1,20 +1,26 @@
 import { onError } from "../utils/response";
 import validation, { updateValidation } from "../validators/trip";
-
+import { roles } from '../helpers/roles'
 export default class Trip {
-  static validate(req, res, next) {
-    const auth = validation(req.body);
-    if (auth.error) {
-      return onError(res, 400, auth.error.details[0].message);
+    static validate(req, res, next) {
```
Comment (niyodusengaclement) [inline src/middlewares/trip.js]: This validation should be done in validators! No need for middleware and validations that are working same thing

File: src/middlewares/trip.js [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+        next();
     }
-    next();
-  }
 
-  static validateUpdate(req, res, next) {
-    const { error } = updateValidation(req.body);
-    if (error) {
-      return onError(res, 400, error.details[0].message);
+    static validateUpdate(req, res, next) {
```
Comment (niyodusengaclement) [inline src/middlewares/trip.js:13]: This validation should be done in validators! No need for middleware and validations that are working the same thing

File: src/controllers/trip.js [hunk 3]
```diff
@@ -1,63 +1,67 @@
...
+            ],
+        });
+        if (!trip) {
+            return res.status(404).json({ status: 404, message: "Trip not found" });
+        }
+        if (
+            trip.requester_id !== req.user.id
+            // || trip.manager_id !== req.user.manager_id
+        ) {
+            return res.status(403).json({ status: 403, message: "Trip not yours!" });
```
Comment (niyodusengaclement) [inline src/controllers/trip.js]: No need for repeating all of these response requirements as long as we have a response helper. Use onError or onSuccess to return response to the user

File: src/controllers/trip.js [hunk 4]
```diff
@@ -1,63 +1,67 @@
...
-    //   return res
-    //     .status(403)
-    //     .json({ status: 403, error: "unauthorized to check all trips" });
-    // }
 
-    const trips = await Trips.findAll({
-      where: { manager_id: req.user.id },
-    });
+    static async getAll(req, res) {
+        // if (req.user.role !== "manager") {
```
Comment (niyodusengaclement) [inline src/controllers/trip.js]: No comments

**Review by niyodusengaclement — CHANGES_REQUESTED** [review]
Fix these issues

**Review by niyodusengaclement — APPROVED** [review]

**Review by RedJanvier — APPROVED** [review]

**Pull request merged**